src/frontend/src/eth/components/wallet-connect/WalletConnectButton.svelte

```diff
@@ -1,9 +1,17 @@
 <script lang="ts">
 	import IconWalletConnect from '$lib/components/icons/IconWalletConnect.svelte';
 	import { ethAddressNotLoaded } from '$lib/derived/address.derived';
+
+	export let ariaLabel: string | undefined = undefined;
 </script>
 
-<button on:click class="tertiary text-primary h-10" disabled={$ethAddressNotLoaded}>
+<button
```

Comment:
maybe some transition, to make the connection change smoother?

Reply:
Can you help me out here?

What transition were you thinking of? I think if we want to have a smooth transition, I would have to change the usage in WalletConnectSession.svelte to render the same button, but with different content / onClick handlers?

Beginning of the video i just switch between connected / disconnected to show the transition fade from svelte.
End of the video is the actual implementation with the connect modal.

https://github.com/user-attachments/assets/a85cc334-5a08-40a5-bc0e-706fb9784aee

Reply:
I would first try one thing:

what if the button is just one and you change the props/labels based on a condition?

```svelte
<WalletConnectButton
	on:click={nonNullish(listener) ? disconnect : openWalletConnectAuth}
	ariaLabel={nonNullish(listener) ? $i18n.wallet_connect.text.name : the_other_arialabel}
>
	{#if nonNullish(listener)}
		<span transition:slide={SLIDE_PARAMS}>{$i18n.wallet_connect.text.disconnect}</span>
	{/if}
</WalletConnectButton>
```


I am not 100% sure if it is better or not, though, or if it fits the requirements

Reply:
Had to tried different things and ended up with:
https://github.com/user-attachments/assets/fa33f871-e012-4e0b-9761-7a0a47caf139

Reply:
thank you!